src/Hook.php

```diff
@@ -22,10 +22,19 @@ public static function getValidHooks()
         );
         $validHooks = [];
 
+        //add customize hooks
+        $customizeHooks = isset($json['extra']['customize_hooks']) ?
```

Comment:
Actually, git hooks are more than just scripts. They are tied into git's event system and run only when an event occurs, so custom ones don't really make sense. A [read](https://git-scm.com/book/gr/v2/Customizing-Git-Git-Hooks) about git hooks.

Anyway, I would appreciate it if you could remove that commit for now, since it's not really tied to the original feature you were implementing. Then I can merge, and we can talk about the custom hooks in a separate PR. Thanks